packages/drivers/odsp-driver/src/odspDocumentStorageManager.ts

```diff
@@ -459,12 +352,261 @@ export class OdspDocumentStorageService implements IDocumentStorageService {
+                // after the main container). In this case, we do not need to do any hashing
+                if (path) {
+                    // Schedule the hashes for later, but keep track of the tasks
+                    // to ensure they finish before they might be used
+                    const hashP = hashFile(IsoBuffer.from(blob.content, blob.encoding)).then((hash: string) => {
+                        this.blobsShaToPathCache.set(hash, path);
+                    });
+                    this.blobsCachePendingHashes.add(hashP);
+                    // eslint-disable-next-line @typescript-eslint/no-floating-promises
+                    hashP.finally(() => {
+                        this.blobsCachePendingHashes.delete(hashP);
+                    });
+                }
+            }
+        }
+
+        this.ops = ops;
+        return id ? [{ id, treeId: undefined! }] : [];
+    }
+
+    private async fetchSnapshotWithMultiPartRequestCore(snapshotOptions: ISnapshotOptions, tokenFetchOptions: TokenFetchOptions) {
         const storageToken = await this.getStorageToken(tokenFetchOptions, "TreesLatest");
 
         // TODO: This snapshot will return deltas, which we currently aren't using. We need to enable this flag to go down the "optimized"
         // snapshot code path. We should leverage the fact that these deltas are returned to speed up the deltas fetch.
-        const { headers, url } = getUrlAndHeadersWithAuth(`${this.snapshotUrl}/trees/latest${snapshotOptions}`, storageToken);
+        const url = `${this.snapshotUrl}/trees/latest?ump=1`;
+        const formBoundary = uuid();
+        const crlf = "\r\n";
+        let postBody = `--${formBoundary}${crlf}Authorization: Bearer ${storageToken}${crlf}X-HTTP-Method-Override: GET${crlf}`;
+        Object.entries(snapshotOptions).forEach(([key, value]) => {
+            postBody += `${key}: ${value}${crlf}`;
+        });
+        postBody += `${crlf}--${formBoundary}--`;
+        const headers = {
+            "Content-Type": `multipart/form-data;boundary=${formBoundary}`,
+        };
+        // This event measures only successful cases of getLatest call (no tokens, no retries).
+        const { snapshot, canCache } = await PerformanceEvent.timedExecAsync(this.logger, { eventName: "TreesLatest" }, async (event) => {
+            const response = await fetchHelper<IOdspSnapshot>(
+                url,
+                {
+                    body: postBody,
+                    headers,
+                    method: "POST",
+                });
+            const content = response.content;
+            event.end({
+                trees: content.trees?.length ?? 0,
+                blobs: content.blobs?.length ?? 0,
+                ops: content.ops?.length ?? 0,
+                headers: Object.keys(headers).length !== 0 ? true : undefined,
+                sprequestguid: response.headers.get("sprequestguid"),
```

Comment:
Looks like some of Raj's w3c telemetry should be added here?

Reply:
I will merge this after his PR. So will pull his changes.